pkg/metrics/collector.go

```diff
@@ -96,21 +102,254 @@ func (c *metricsCollector) publishVolumeStats(ctx context.Context, volume *types
 	)
 }
 
+type DriveStats struct {
+	ReadSectors  int64
+	ReadTicks    int64
+	WriteSectors int64
+	WriteTicks   int64
+	TimeInQueue  int64
+}
+
+func (c *metricsCollector) publishDriveStats(ctx context.Context, drive *types.Drive, ch chan<- prometheus.Metric) {
+	device, err := c.getDeviceByFSUUID(drive.Status.FSUUID)
+	if err != nil {
+		klog.ErrorS(
+			err,
+			"unable to find device by FSUUID; "+
+				"either device is removed or run command "+
+				"`sudo udevadm control --reload-rules && sudo udevadm trigger`"+
+				" on the host to reload",
+			"FSUUID", drive.Status.FSUUID)
+		client.Eventf(
+			drive, client.EventTypeWarning, client.EventReasonMetrics,
+			"unable to find device by FSUUID %v; "+
+				"either device is removed or run command "+
+				"`sudo udevadm control --reload-rules && sudo udevadm trigger`"+
+				" on the host to reload", drive.Status.FSUUID)
+
+		return
+	}
+
+	// Online/Offline Status
+	if _, err := os.Stat("/sys/block/" + device); os.IsNotExist(err) {
```

Comment:
This will not work for partitions. The partitions info will be under `/sys/block/<parent-dev>/<dev>/`

Reply:
`device, err := c.getDeviceByFSUUID(drive.Status.FSUUID)` returns the device in this format `/dev/dm-1`, so there's no way access the partitions. in my case `dm-0` and `dm-1` are both drives.

Reply:
Ignore this.. `/sys/block/<dev>/stat` will work. There are no any special cases for partitions. I see we have symlinks appropriately.